src/components/swaptab/swaptabwrapper.js

```diff
@@ -232,43 +280,64 @@ class SwapTabWrapper extends React.Component {
     const rate = this.state.rate;
 
     if (rate) {
-      return rate.rate.toFixed(5);
+      const exactRate = new BigNumber(rate.rate);
+      return exactRate.toFixed(5);
     } else {
       return 'Not found';
     }
   };
 
   switchPair = () => {
-    this.setState({
-      base: this.state.quote,
-      baseAmount: this.state.quoteAmount,
-      quote: this.state.base,
-      quoteAmount: this.state.baseAmount,
-    });
+    if (this.state.rate) {
+      this.setState(
+        {
+          base: this.state.quote,
+          quote: this.state.base,
+        },
+        () => {
+          const symbol = this.getSymbol();
+          const limits = this.props.limits[symbol].minimal / decimals;
```

Comment:
Why use the minimal limit as default after switching the assets? Getting the quote amount, setting it as new base amount and `updateQuoteAmount` should work fine.